### PR TITLE
Issue #9492: updated example of AST for TokenTypes.LITERAL_RETURN

### LINKED ISSUE #9492: update example of AST for TokenTypes.LITERAL_RETURN
https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_RETURN

needs to be extended to show example of AST.
Source to be updated: https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java

PLEASE READ for more details: https://github.com/checkstyle/checkstyle/wiki/Good-practice-on-how-to-update-AST-in-TokenTypes

### PR DESCRIPTION
**Pull request by Devansu-Yadav**
Closes #9492 
CheckStyle_return_changed_PR_photo_2

```
$ cat Test.java

public class Test{
    public int foo(int i) {
        return i+1;
    }
}
```
AST in CLI for the above Java file:-
```
$ java -jar checkstyle-8.41-all.jar -T Test.java
CLASS_DEF -> CLASS_DEF [1:0]
|--MODIFIERS -> MODIFIERS [1:0]
|   `--LITERAL_PUBLIC -> public [1:0]
|--LITERAL_CLASS -> class [1:7]
|--IDENT -> Test [1:13]
`--OBJBLOCK -> OBJBLOCK [1:17]
    |--LCURLY -> { [1:17]
    |--METHOD_DEF -> METHOD_DEF [2:4]
    |   |--MODIFIERS -> MODIFIERS [2:4]
    |   |   `--LITERAL_PUBLIC -> public [2:4]
    |   |--TYPE -> TYPE [2:11]
    |   |   `--LITERAL_INT -> int [2:11]
    |   |--IDENT -> foo [2:15]
    |   |--LPAREN -> ( [2:18]
    |   |--PARAMETERS -> PARAMETERS [2:19]
    |   |   `--PARAMETER_DEF -> PARAMETER_DEF [2:19]
    |   |       |--MODIFIERS -> MODIFIERS [2:19]
    |   |       |--TYPE -> TYPE [2:19]
    |   |       |   `--LITERAL_INT -> int [2:19]
    |   |       `--IDENT -> i [2:23]
    |   |--RPAREN -> ) [2:24]
    |   `--SLIST -> { [2:26]
    |       |--LITERAL_RETURN -> return [3:8]
    |       |   |--EXPR -> EXPR [3:16]
    |       |   |   `--PLUS -> + [3:16]
    |       |   |       |--IDENT -> i [3:15]
    |       |   |       `--NUM_INT -> 1 [3:17]
    |       |   `--SEMI -> ; [3:18]
    |       `--RCURLY -> } [4:4]
    `--RCURLY -> } [5:0]
```
```
# printing lines we are concerned with:-
$ java -jar checkstyle-8.41-all.jar -T Test.java | grep "2:\|3:\|4:"

|--METHOD_DEF -> METHOD_DEF [2:4]
|   |--MODIFIERS -> MODIFIERS [2:4]
|   |   `--LITERAL_PUBLIC -> public [2:4]
|   |--TYPE -> TYPE [2:11]
|   |   `--LITERAL_INT -> int [2:11]
|   |--IDENT -> foo [2:15]
|   |--LPAREN -> ( [2:18]
|   |--PARAMETERS -> PARAMETERS [2:19]
|   |   `--PARAMETER_DEF -> PARAMETER_DEF [2:19]
|   |       |--MODIFIERS -> MODIFIERS [2:19]
|   |       |--TYPE -> TYPE [2:19]
|   |       |   `--LITERAL_INT -> int [2:19]
|   |       `--IDENT -> i [2:23]
|   |--RPAREN -> ) [2:24]
|   `--SLIST -> { [2:26]
|       |--LITERAL_RETURN -> return [3:8]
|       |   |--EXPR -> EXPR [3:16]
|       |   |   `--PLUS -> + [3:16]
|       |   |       |--IDENT -> i [3:15]
|       |   |       `--NUM_INT -> 1 [3:17]
|       |   `--SEMI -> ; [3:18]
|       `--RCURLY -> } [4:4]
```
Expected update for JavaDoc:-
```
METHOD_DEF -> METHOD_DEF 
 |--MODIFIERS -> MODIFIERS 
 |   `--LITERAL_PUBLIC -> public 
 |--TYPE -> TYPE 
 |   `--LITERAL_INT -> int 
 |--IDENT -> foo 
 |--LPAREN -> ( 
 |--PARAMETERS -> PARAMETERS 
 |   `--PARAMETER_DEF -> PARAMETER_DEF 
 |       |--MODIFIERS -> MODIFIERS 
 |       |--TYPE -> TYPE 
 |       |   `--LITERAL_INT -> int 
 |       `--IDENT -> i 
 |--RPAREN -> ) 
 `--SLIST -> { 
     |--LITERAL_RETURN -> return 
     |   |--EXPR -> EXPR 
     |   |   `--PLUS -> + 
     |   |       |--IDENT -> i 
     |   |       `--NUM_INT -> 1 
     |   `--SEMI -> ; 
     `--RCURLY -> } 
```

### CODE REVIEW

**Comment by Devansu-Yadav** [comment]
@timurt @romani Can u pls review this PR?

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java [hunk 1]
```diff
@@ -2384,6 +2384,31 @@
      * optional expression for the return value.  The last child is a
      * semi colon.
      *
+     * <p>For example:</p>
+     * <pre>
+     * {@code
```
Comment (timurt) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:2389]: you do not need to use `{@code }`

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java [hunk 2]
```diff
@@ -2384,6 +2384,31 @@
...
+     * public void foo(){
```
Comment (timurt) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:2390]: missing whitespace before curly braces

**Review by timurt — CHANGES_REQUESTED** [review]

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java (hunk 1 above)
Comment (Devansu-Yadav) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java]: Done!

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java (hunk 2 above)
Comment (Devansu-Yadav) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java]: Done!

**Comment by romani** [comment]
Please make example as:
```
public int foo(int i) {
        return i + 1;
    }
```

**Comment by Devansu-Yadav** [comment]
@romani Done! Updated PR description as well.

**Review by romani — APPROVED** [review]

**Pull request merged**